pkg/osbuild/curl_source.go

```diff
@@ -3,14 +3,47 @@ package osbuild
 import (
 	"bytes"
 	"encoding/json"
+	"regexp"
+
+	"github.com/osbuild/images/pkg/rpmmd"
 )
 
+var curlDigestPattern = regexp.MustCompile(`(md5|sha1|sha256|sha384|sha512):[0-9a-f]{32,128}`)
+
 type CurlSource struct {
 	Items map[string]CurlSourceItem `json:"items"`
 }
 
 func (CurlSource) isSource() {}
 
+func NewCurlSource() *CurlSource {
+	return &CurlSource{
+		Items: make(map[string]CurlSourceItem),
+	}
+}
+
+func NewCurlPackageItem(pkg rpmmd.PackageSpec) CurlSourceItem {
+	if !curlDigestPattern.MatchString(pkg.Checksum) {
+		panic("curl package source item has invalid digest")
```

Comment:
I'm not convinced that using `panic()` is the right approach here. The digest hash comes from depsolving using DNF and it is AFAIK not hardcoded in the code. So it may as well happen that composer will get unsupported hash algorithm, but there is in fact no programming error in composer.

Returning an `error` and failing the image build with sensible error message seems like a more reasonable approach to me.

I think that this is a special case, compared to other stages, where we really want to panic.

Reply:
You make a great point.  I'll change it.

Reply:
Done!